src/Microsoft.OData.Core/Value/ODataJsonElementValue.cs

```diff
@@ -0,0 +1,39 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataJsonElementValue.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETCOREAPP3_1_OR_GREATER
+using System.Text.Json;
+
+namespace Microsoft.OData
+{
+    /// <summary>
+    /// Represents a <see cref="JsonElement"/> value.
+    /// Useful for writing parsed  JSON inputs directly without reading them first, which would result in double allocations.
+    /// </summary>
+    public sealed class ODataJsonElementValue : ODataValue
+    {
+        /// <summary>
+        /// Creates an intsance of <see cref="ODataJsonElementValue"/> based
+        /// on the specified <see cref="JsonElement"/> value.
+        /// </summary>
+        /// <param name="value">The <see cref="JsonElement"/> to wrap.</param>
+        /// <remarks>
+        /// The caller is responsible for ensuring the contents of the <see cref="JsonElement"/> are valid

```

Comment:
what happens if the caller does not ensure the validity of contents? Will ODataWriter just write them?

Reply:
Yes, it will write whatever is in the `JsonElement` without checking it.